semantic_model_generator/tests/generate_sql_test.py

```diff
@@ -0,0 +1,94 @@
+import pytest
+
+from semantic_model_generator.protos.semantic_model_pb2 import (
+    AggregationType,
+    Dimension,
+    FullyQualifiedTable,
+    Measure,
+    Table,
+    TimeDimension,
+)
+from semantic_model_generator.sqlgen.generate_sql import generate_select_with_all_cols
+
+dimension_example = Dimension(
+    name="Region",
+    synonyms=["Area", "Locale"],
+    description="Geographical region",
+    expr="region_code",
+    data_type="string",
+    unique=False,
+    sample_values=["North", "South", "East", "West"],
+)
+
+dimension_example_no_cols = Dimension(
+    synonyms=["Area", "Locale"],
+    description="Geographical region",
+)
+
+time_dimension_example = TimeDimension(
+    name="Date",
+    synonyms=["Time"],
+    description="Transaction date",
+    expr="transaction_date",
+    data_type="date",
+    unique=False,
+    sample_values=["2022-01-01", "2022-01-02", "2022-01-03"],
+)
+
+measure_example = Measure(
+    name="Total Sales",
```

Comment:
We should stick to names without spaces for now - they'll work here but we don't properly handle them downstream.